### PR TITLE
Account Statictics own visit added after every visit

### LINKED ISSUE #5311: [BUG] Account Statistics - own visits added after every refresh
### Description

Currently my own visits to my own profile and as well the /account/statistics page adding towards to view count in statistics.

This happens when
- going to the profile/statistics via any link
- on refresh

I can do it multiple times and always the view count gets up.

In my opinion at least my own views shouldn't ve counted. Especially if I'm visiting the sratistics page.

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Akhilbisht798**
## Fixes Issue

Closes #5311 

## Changes proposed

- [x] Added to check if auth user and profile are same if they are don't increment the counter.

## Check List (Check all the applicable boxes) 

- [X] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] This PR does not contain plagiarized content.
- [X] The title of my pull request is a short description of the requested changes.

## Note to reviewers

This is not completed. This is a draft pr.

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/5595"></a>

### CODE REVIEW

**Comment by Akhilbisht798** [comment]
This is not fixed. I have made a draft pr to let other see my code and what could be wrong with this.

File: pages/api/users/[username]/index.js [hunk 1]
```diff
@@ -14,12 +17,16 @@ export default async function handler(req, res) {
       .status(400)
       .json({ error: "Invalid request: GET request required" });
   }
-
-  const { status, profile } = await getUserApi(req.query.username);
+  let sameUser = false;
+  const session = await unstable_getServerSession(req, res, authOptions);
```
Comment (eddiejaoude) [inline pages/api/users/[username]/index.js:21]: This needs to happen in the `getUserApi` function, because the UI calls that function directly

Any API calls will also call that function via this `handler` function

Comment (eddiejaoude) [reply pages/api/users/[username]/index.js:21]: I think you can pass in `req` and `res` in the file `pages/[username].js` and update `getUserApi` to use them as parameters

Comment (eddiejaoude) [reply pages/api/users/[username]/index.js:27]:
```diff
diff --git a/pages/[username].js b/pages/[username].js
index 3f26a95b..cf561e03 100644
--- a/pages/[username].js
+++ b/pages/[username].js
@@ -15,14 +15,14 @@ import Page from "@components/Page";
 import UserPage from "@components/user/UserPage";
 
 export async function getServerSideProps(context) {
-  const { req } = context;
+  const { req, res } = context;
   const username = context.query.username;
   const log = logger.child({
     username: username,
     ip: requestIp.getClientIp(req),
   });
 
-  const { status, profile } = await getUserApi(username);
+  const { status, profile } = await getUserApi(req, res, username);
   if (status !== 200) {
     log.error(
       profile.error,
diff --git a/pages/api/users/[username]/index.js b/pages/api/users/[username]/index.js
index 56780ec1..9f088e32 100644
--- a/pages/api/users/[username]/index.js
+++ b/pages/api/users/[username]/index.js
@@ -1,5 +1,7 @@
 import connectMongo from "@config/mongo";
 import logger from "@config/logger";
+import { authOptions } from "../../auth/[...nextauth]";
+import { unstable_getServerSession } from "next-auth/next";
 
 import Profile from "@models/Profile";
 import Stats from "@models/Stats";
@@ -15,11 +17,14 @@ export default async function handler(req, res) {
       .json({ error: "Invalid request: GET request required" });
   }
 
-  const { status, profile } = await getUserApi(req.query.username);
+  const { status, profile } = await getUserApi(req, res, req.query.username);
   return res.status(status).json(profile);
 }
 
-export async function getUserApi(username) {
+export async function getUserApi(req, res, username) {
+  const session = await unstable_getServerSession(req, res, authOptions);
+  console.log(session);
+
   await connectMongo();
 
   const log = logger.child({ username: username });
```

**Comment by Akhilbisht798** [comment]
Completed This pr. Ready for review

File: pages/api/users/[username]/index.js (hunk 1 above)
Comment (Akhilbisht798) [reply pages/api/users/[username]/index.js:27]: Made all the changes. Ready for review

File: pages/account/statistics.js [hunk 2]
```diff
@@ -38,7 +38,12 @@ export async function getServerSideProps(context) {
   let profile = {};
   try {
     const resUser = await fetch(
-      `${process.env.NEXT_PUBLIC_BASE_URL}/api/users/${username}`
+      `${process.env.NEXT_PUBLIC_BASE_URL}/api/users/${username}`,
+      {
+        headers: {
+          cookie: context.req.headers.cookie || "",
+        },
+      }
```
Comment (eddiejaoude) [inline pages/account/statistics.js:46]: Why is this needed?

File: pages/api/users/[username]/index.js [hunk 3]
```diff
@@ -14,12 +17,17 @@ export default async function handler(req, res) {
...
   return res.status(status).json(profile);
 }
 
-export async function getUserApi(username) {
+export async function getUserApi(req, res, username) {
+  let sameUser = false;
+  const session = await unstable_getServerSession(req, res, authOptions);
+  if (session) {
+    sameUser = (session.username === username) ? true : false;
+  }
```
Comment (eddiejaoude) [inline pages/api/users/[username]/index.js:30]: I think as it is already false the extra check is not needed

```suggestion
  if (session && session.username === username) {
    sameUser = true;
  }
```

I will fix via a temporary branch

File: pages/account/statistics.js (hunk 2 above)
Comment (eddiejaoude) [reply pages/account/statistics.js:46]: Does it not work without it?

Comment (eddiejaoude) [reply pages/account/statistics.js:46]: I removed it and it worked fine for me

**Comment by eddiejaoude** [comment]
Thank you 👍

**Comment by Akhilbisht798** [comment]
Thank you Eddie for the help and mentor ship that you have given me in this pr. This was my first pr which was big and requires some large changes to the code base. I really appreciate your help and this was fun and I will  try my best to contribute even more now.

**Comment by eddiejaoude** [comment]
Great work, keep going @Akhilbisht798 !
